src/BlockInfo.cpp

```diff
@@ -838,43 +838,13 @@ bool cBlockInfo::FullyOccupiesVoxel(const BLOCKTYPE Block)
 
 bool cBlockInfo::IsOneHitDig(const BLOCKTYPE Block)
 {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wfloat-equal"
+
 	// One hit break blocks:
-	switch (Block)
-	{
-		case E_BLOCK_ACTIVE_COMPARATOR:
-		case E_BLOCK_BEETROOTS:
-		case E_BLOCK_BIG_FLOWER:
-		case E_BLOCK_BROWN_MUSHROOM:
-		case E_BLOCK_CARROTS:
-		case E_BLOCK_CROPS:
-		case E_BLOCK_DANDELION:
-		case E_BLOCK_DEAD_BUSH:
-		case E_BLOCK_END_ROD:
-		case E_BLOCK_FIRE:
-		case E_BLOCK_FLOWER:
-		case E_BLOCK_FLOWER_POT:
-		case E_BLOCK_INACTIVE_COMPARATOR:
-		case E_BLOCK_LILY_PAD:
-		case E_BLOCK_MELON_STEM:
-		case E_BLOCK_NETHER_WART:
-		case E_BLOCK_POTATOES:
-		case E_BLOCK_PUMPKIN_STEM:
-		case E_BLOCK_REDSTONE_REPEATER_OFF:
-		case E_BLOCK_REDSTONE_REPEATER_ON:
-		case E_BLOCK_REDSTONE_TORCH_OFF:
-		case E_BLOCK_REDSTONE_TORCH_ON:
-		case E_BLOCK_REDSTONE_WIRE:
-		case E_BLOCK_RED_MUSHROOM:
-		case E_BLOCK_REEDS:
-		case E_BLOCK_SAPLING:
-		case E_BLOCK_SLIME_BLOCK:
-		case E_BLOCK_TNT:
-		case E_BLOCK_TALL_GRASS:
-		case E_BLOCK_TORCH:
-		case E_BLOCK_TRIPWIRE_HOOK:
-		case E_BLOCK_TRIPWIRE: return true;
-		default: return false;
-	}
+	return GetHardness(Block) == 0;
```

Comment:
This can be simplified
```suggestion
	return !GetHardness(Block);
```

see http://cpp.sh/4w4sm3

Reply:
although I didn't check if this also triggers clang. I'm thinking about a simple idea to get rid of the clang pacifier

Reply:
It triggers another warning (conversion float to bool).

I think the warning is bogus here; we're able to compare to exactly floating zero because GetHardness returns exactly zero.

Reply:
Yeah - you are right you actually set it to 0. Maybe leave a comment?

Reply:
OK, fixed.